client/document.go

```diff
@@ -118,11 +117,11 @@ func NewDocFromMap(data map[string]any, collectionDefinition CollectionDefinitio
 	return doc, nil
 }
 
-var jsonArrayPattern = regexp.MustCompile(`^\s*\[.*\]\s*$`)
-
 // IsJSONArray returns true if the given byte array is a JSON Array.
 func IsJSONArray(obj []byte) bool {
-	return jsonArrayPattern.Match(obj)
+	var js []any
+	err := json.Unmarshal(obj, &js)
+	return err == nil
```

Comment:
question: What made you do this?

Reply:
this test wouldn't pass with original implementation https://github.com/sourcenetwork/defradb/pull/2731/files#diff-8c5dc24d796c5772fcbb82cb58fdfd8e6046ca461d3fae69bf934f67d21dac25R252-R257

Reply:
Is it because the pattern approach didn't account for a JSON array with whitespace?

Reply:
Use of `json.Unmarshal` here makes me feel like it's not that efficient as it probably uses reflect.

How about something like: 

```suggestion
	trimmedObj := bytes.TrimLeft(obj, " \r\t\n")
	return len(trimmedObj) > 0 && trimmedObj[0] == '['
```

Reply:
I investigated some more why regexp didn't work for this case and found a good fix.
Thanks for pointing that out.